libcloud/utils/misc.py

```diff
@@ -282,3 +292,28 @@ def __repr__(self):
 
     def __str__(self):
         return str(self.__repr__())
+
+
+def retry(timeout, retry_delay, backoff, func, *args, **kwargs):
+    """
+    Retry decorator helps to handle common exception.
+    """
+    timeout = DEFAULT_TIMEOUT if timeout is None else timeout
```

Comment:
We originally implemented it as a decorator, but it made testing difficult because the timing values were all set at compile-time and you couldn't override them to make it re-run in a tighter loop in the tests, causing the test time to blow up considerably.  Is there a good solution for that problem?  The only solution I'm aware of is to make sure you set the default values before you load any code that uses the decorator, which is hard to guarantee.